### ISSUE TITLE
Async functions with native generators

### ISSUE THREAD

**Issue by zloirock** [6to5/6to5#321]
Generators now avaliable on Node.js, but 6to5 uses regenerator for async functions. You can add fallback for async functions without regenerator:
```javascript
async function(){/* */} -> __async(function*(){/* */})
await -> yield

function __async(genF){
  return function(){
    var gen = genF.apply(this, arguments);
    return new Promise(function(resolve, reject){
      function step(nextF){
        var next;
        try {
          next = nextF();
        } catch(e){
          reject(e);
          return;
        }
        if(next.done){
          resolve(next.value);
          return;
        }
        Promise.resolve(next.value).then(function(v){
          step(function(){return gen.next(v)});
        }, function(e){
          step(function(){return gen['throw'](e)});
        });
      }
      step(function(){return gen.next()});
    });
  }
}
```

Helper based on [this](https://github.com/lukehoban/ecmascript-asyncawait#spawning). [Basic example](http://goo.gl/rksdJx). [Example with this transform](http://goo.gl/tIjF0V).

**Comment by zloirock** [comment]
:thumbsup:

**Comment by zloirock** [comment]
Problem with named functions: `async function f(){}` -> `_asyncToGenerator(function* f() {})`

**Comment by sebmck** [comment]
@zloirock Thanks, that was trickier than I thought due to avoiding hoisting above the `require` calls. Fixed as of 2.4.3.

**Comment by zloirock** [comment]
Awesome!
